File: CustomUI/app/src/main/java/com/pdftron/android/tutorial/customui/MainActivity.java
package com.pdftron.android.tutorial.customui;

import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;

import com.pdftron.pdf.Annot;
import com.pdftron.pdf.config.ToolManagerBuilder;
import com.pdftron.pdf.config.ViewerBuilder2;
import com.pdftron.pdf.config.ViewerConfig;
import com.pdftron.pdf.controls.PdfViewCtrlTabHostFragment;
import com.pdftron.pdf.controls.PdfViewCtrlTabHostFragment2;
import com.pdftron.pdf.model.FileInfo;
import com.pdftron.pdf.tools.QuickMenu;
import com.pdftron.pdf.tools.QuickMenuItem;
import com.pdftron.pdf.tools.ToolManager;
import com.pdftron.pdf.utils.Utils;
import com.pdftron.pdf.widget.toolbar.builder.AnnotationToolbarBuilder;
import com.pdftron.pdf.widget.toolbar.builder.ToolbarButtonType;
import com.pdftron.pdf.widget.toolbar.component.DefaultToolbars;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements PdfViewCtrlTabHostFragment.TabHostListener {

    private PdfViewCtrlTabHostFragment2 mPdfViewCtrlTabHostFragment;

    public static final String NOTES_TOOLBAR_TAG = "notes_toolbar";
    public static final String SHAPES_TOOLBAR_TAG = "shapes_toolbar";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Instantiate a PdfViewCtrlTabHostFragment with a document Uri
        File f = Utils.copyResourceToLocal(this, R.raw.sample, "sample", ".pdf");
        Uri uri = Uri.fromFile(f);
        ToolManagerBuilder tmBuilder = ToolManagerBuilder.from()
                .disableToolModes(new ToolManager.ToolMode[] {
                        ToolManager.ToolMode.AREA_MEASURE_CREATE,
                        ToolManager.ToolMode.PERIMETER_MEASURE_CREATE,
                        ToolManager.ToolMode.RECT_AREA_MEASURE_CREATE,
                        ToolManager.ToolMode.RULER_CREATE
                });
        ViewerConfig viewerConfig = new ViewerConfig.Builder()
                .addToolbarBuilder(buildNotesToolbar())
                .addToolbarBuilder(buildShapesToolbar())
                .toolbarTitle("٩(◕‿◕｡)۶")
                .toolManagerBuilder(tmBuilder)
                .build();
        mPdfViewCtrlTabHostFragment = ViewerBuilder2.withUri(uri)
                .usingCustomToolbar(new int[] {R.menu.my_custom_options_toolbar})
                .usingNavIcon(R.drawable.ic_star_white_24dp)
                .usingConfig(viewerConfig)
                .usingTheme(R.style.CustomAppTheme)
                .build(this);
        mPdfViewCtrlTabHostFragment.addHostListener(this);

        // Apply customizations to tab host fragment
//        new CustomQuickMenu(MainActivity.this, mPdfViewCtrlTabHostFragment);
//        new CustomLinkClick(MainActivity.this, mPdfViewCtrlTabHostFragment);
//        new CustomAnnotationToolbar(MainActivity.this, mPdfViewCtrlTabHostFragment);

        // Add the fragment to our activity
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.fragment_container, mPdfViewCtrlTabHostFragment);
        ft.commit();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mPdfViewCtrlTabHostFragment != null) {
            mPdfViewCtrlTabHostFragment.removeHostListener(this);
        }
    }

    private AnnotationToolbarBuilder buildNotesToolbar() {
        return AnnotationToolbarBuilder.withTag(NOTES_TOOLBAR_TAG) // Identifier for toolbar
                .setToolbarName("Notes Toolbar") // Name used when displaying toolbar
                .addToolButton(ToolbarButtonType.INK, 1)
                .addToolButton(ToolbarButtonType.STICKY_NOTE, 2)
                .addToolButton(ToolbarButtonType.TEXT_HIGHLIGHT, 3)
                .addToolButton(ToolbarButtonType.TEXT_UNDERLINE, 4)
                .addToolButton(ToolbarButtonType.TEXT_STRIKEOUT, 5)
                .addToolStickyButton(ToolbarButtonType.UNDO, DefaultToolbars.ButtonId.UNDO.value())
                .addToolStickyButton(ToolbarButtonType.REDO, DefaultToolbars.ButtonId.REDO.value());
    }


    private AnnotationToolbarBuilder buildShapesToolbar() {
        return AnnotationToolbarBuilder.withTag(SHAPES_TOOLBAR_TAG) // Identifier for toolbar
                .setToolbarName("Shapes Toolbar") // Name used when displaying toolbar
                .addToolButton(ToolbarButtonType.SQUARE, DefaultToolbars.ButtonId.SQUARE.value())
                .addToolButton(ToolbarButtonType.CIRCLE, DefaultToolbars.ButtonId.CIRCLE.value())
                .addToolButton(ToolbarButtonType.LINE, DefaultToolbars.ButtonId.LINE.value())
                .addToolButton(ToolbarButtonType.POLYGON, DefaultToolbars.ButtonId.POLYGON.value())
                .addToolButton(ToolbarButtonType.POLYLINE, DefaultToolbars.ButtonId.POLYLINE.value())
                .addToolStickyButton(ToolbarButtonType.UNDO, DefaultToolbars.ButtonId.UNDO.value())
                .addToolStickyButton(ToolbarButtonType.REDO, DefaultToolbars.ButtonId.REDO.value());
    }

    @Override
    public void onTabDocumentLoaded(String s) {
        mPdfViewCtrlTabHostFragment.getCurrentPdfViewCtrlFragment().addQuickMenuListener(new ToolManager.QuickMenuListener() {
            @Override
            public boolean onQuickMenuClicked(QuickMenuItem menuItem) {
                return false;
            }

            @Override
            public boolean onShowQuickMenu(QuickMenu quickmenu, @Nullable Annot annot) {
                ArrayList<QuickMenuItem> removeList = new ArrayList<>();
                checkQuickMenu(quickmenu.getFirstRowMenuItems(), removeList, R.id.qm_search);
                checkQuickMenu(quickmenu.getFirstRowMenuItems(), removeList, R.id.qm_share);
                checkQuickMenu(quickmenu.getSecondRowMenuItems(), removeList, R.id.qm_search);
                checkQuickMenu(quickmenu.getSecondRowMenuItems(), removeList, R.id.qm_share);
                checkQuickMenu(quickmenu.getOverflowMenuItems(), removeList, R.id.qm_search);
                checkQuickMenu(quickmenu.getOverflowMenuItems(), removeList, R.id.qm_share);
                quickmenu.removeMenuEntries(removeList);
                return false;
            }

            @Override
            public void onQuickMenuShown() {

            }

            @Override
            public void onQuickMenuDismissed() {

            }
        });
    }

    private void checkQuickMenu(List<QuickMenuItem> menuItems, List<QuickMenuItem> removeList, int removeId) {
        for (QuickMenuItem item : menuItems) {
            int menuId = item.getItemId();
            if (menuId == removeId) {
                removeList.add(item);
            }
        }
    }

    @Override
    public boolean onToolbarOptionsItemSelected(MenuItem menuItem) {
        if (menuItem.getItemId() == R.id.action_show_toast) {
            Toast.makeText(this, "Show toast is clicked!", Toast.LENGTH_SHORT).show();
        }
        return false;
    }

    @Override
    public void onTabHostShown() {

    }

    @Override
    public void onTabHostHidden() {

    }

    @Override
    public void onLastTabClosed() {

    }

    @Override
    public void onTabChanged(String s) {

    }

    @Override
    public boolean onOpenDocError() {
        return false;
    }

    @Override
    public void onNavButtonPressed() {

    }

    @Override
    public void onShowFileInFolder(String s, String s1, int i) {

    }

    @Override
    public boolean canShowFileInFolder() {
        return false;
    }

    @Override
    public boolean canShowFileCloseSnackbar() {
        return false;
    }

    @Override
    public boolean onToolbarCreateOptionsMenu(Menu menu, MenuInflater menuInflater) {
        return false;
    }

    @Override
    public boolean onToolbarPrepareOptionsMenu(Menu menu) {
        return false;
    }

    @Override
    public void onStartSearchMode() {

    }

    @Override
    public void onExitSearchMode() {

    }

    @Override
    public boolean canRecreateActivity() {
        return true;
    }

    @Override
    public void onTabPaused(FileInfo fileInfo, boolean b) {

    }

    @Override
    public void onJumpToSdCardFolder() {

    }
}

File: CollabApolloSample/app/src/main/java/com/pdftron/collabapollo/MainActivity.kt
package com.pdftron.collabapollo

import android.content.Context
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.pdftron.collab.client.CollabClient
import com.pdftron.collab.ui.viewer.CollabViewerBuilder2
import com.pdftron.collab.ui.viewer.CollabViewerTabHostFragment2
import com.pdftron.pdf.config.ToolManagerBuilder
import com.pdftron.pdf.config.ViewerConfig

class MainActivity : AppCompatActivity() {

    private val DEFAULT_FILE_NAME = "demo-annotated.pdf"
    private val DEFAULT_FILE_URL =
        "https://pdftron.s3.amazonaws.com/downloads/pl/${DEFAULT_FILE_NAME}"

    private lateinit var mPdfViewCtrlTabHostFragment: CollabViewerTabHostFragment2

    private var mCollabClient = CollabClient.Builder()
        .setDocumentId("12345")
        .setDocumentName(DEFAULT_FILE_NAME)
        .setCollabKey("wv-collab-token")
        .setSecretKey("My_sUper_SEcreT_k3y")
        .setApolloServerIp("192.168.1.167")
        .setApolloServerPort("3000")
        .setAnonymousUsername("testUser")
        .build()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mPdfViewCtrlTabHostFragment =
            createPdfViewerFragment(this, Uri.parse(DEFAULT_FILE_URL), getViewerConfig())
        mPdfViewCtrlTabHostFragment.addCollabHostListener(object :
            CollabViewerTabHostFragment2.CollabTabHostListener {
            override fun onNavButtonPressed() {
                finish()
            }

            override fun onTabDocumentLoaded(p0: String?) {
                handleTabDocumentLoaded(p0!!)
            }

        })

        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.fragment_container, mPdfViewCtrlTabHostFragment, null)
        ft.commit()
    }

    fun handleTabDocumentLoaded(tag: String) {
        mCollabClient.start(
            mPdfViewCtrlTabHostFragment.collabManager,
            mPdfViewCtrlTabHostFragment.currentPdfViewCtrlFragment.pdfViewCtrl
        )
    }

    override fun onDestroy() {
        super.onDestroy()
        mCollabClient.destroy()
    }

    private fun createPdfViewerFragment(
        context: Context,
        fileUri: Uri,
        config: ViewerConfig
    ): CollabViewerTabHostFragment2 {
        return CollabViewerBuilder2
            .withUri(fileUri)
            .usingConfig(getViewerConfig())
            .usingNavIcon(R.drawable.ic_arrow_back_white_24dp)
            .usingTheme(R.style.PDFTronAppTheme)
            .build(this, CollabViewerTabHostFragment2::class.java)
    }

    private fun getViewerConfig(): ViewerConfig {
        val toolManagerBuilder = ToolManagerBuilder.from()
            .setAnnotationLayerEnabled(false)
            .setAutoSelect(true)
        return ViewerConfig.Builder()
            .multiTabEnabled(false)
            .showCloseTabOption(false)
            .saveCopyExportPath(this.filesDir.absolutePath)
            .openUrlCachePath(this.filesDir.absolutePath)
            .toolManagerBuilder(toolManagerBuilder)
            .build()
    }
}